src/main/java/de/tum/in/www1/artemis/service/compass/umlmodel/classdiagram/UMLMethod.java

```diff
@@ -93,6 +98,21 @@ public double similarity(Similarity<UMLElement> reference) {
         return ensureSimilarityRange(similarity);
     }
 
+    /**
+     * Checks if the parent classes of this method and the given reference method are similar/equal by comparing the similarity IDs of both parent classes. If the similarity
+     * IDs are not set, it calculates the similarity of the parent classes itself and checks against the configured equality threshold.
+     *
+     * @param referenceMethod the reference method of which the parent class is compared against the parent class of this method
+     * @return true if the parent classes are similar/equal, false otherwise
+     */
+    private boolean parentsSimilar(UMLMethod referenceMethod) {
+        if (parentClass.getSimilarityID() >= 0 && referenceMethod.getParentClass().getSimilarityID() >= 0) {
```

Comment:
```suggestion
        if (parentClass.getSimilarityID() != -1 && referenceMethod.getParentClass().getSimilarityID() != -1) {
```
I am not sure if my suggested change does make sense. If I see this correctly, the similarityId is -1 if undefined and a in [0,∞[  if defined? If so, I think a `!= =1` check is more explicit.

Reply:
Yes, you are right. I applied your suggested changes.